packages/fxa-auth-server/scripts/refund-unverified-accounts.ts

```diff
@@ -0,0 +1,456 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import program from 'commander';
+import { StripeHelper } from '../lib/payments/stripe';
+import Stripe from 'stripe';
+import Container from 'typedi';
+
+import { setupProcessingTaskObjects } from '../lib/payments/processing-tasks-setup';
+import { AppConfig } from '../lib/types';
+import { promises as fs } from 'fs';
+
+const pckg = require('../package.json');
+
+type SubscriptionData = {
+  subscription: Stripe.Subscription;
+  uid?: string;
+  email?: string;
+  invoicePaymentIntentId?: Stripe.PaymentIntent['id'];
+};
+
+type OutputData = {
+  uid: string;
+  email: string;
+  accountCreated: number;
+  customerId: string;
+  subscriptionId: string;
+  subscriptionCreated: number;
+  subscriptionStatus: string;
+  subscriptionPlan: string;
+  invoicePaymentIntentId: string;
+  refunded: boolean;
+};
+
+type ErrorOutput = {
+  subscriptionId: string;
+  uid: string;
+  email: string;
+  errorMessage: string;
+  error?: string;
+};
+
+class MergedAccountError extends Error {
+  public subscriptionId: string;
+  public uid?: string;
+  public email?: string;
+  constructor(
+    message: string,
+    subscriptionId: string,
+    uid?: string,
+    email?: string
+  ) {
+    super(message);
+    this.message = message;
+    this.subscriptionId = subscriptionId;
+    this.uid = uid;
+    this.email = email;
+  }
+}
+
+const isOfStatus = (
+  keyInput: string
+): keyInput is Stripe.Subscription.Status => {
+  return [
+    'active',
+    'canceled',
+    'incomplete',
+    'incomplete_expired',
+    'past_due',
+    'trialing',
+    'unpaid',
+  ].includes(keyInput);
+};
+
+const isCustomer = (
+  customer: string | Stripe.Customer | Stripe.DeletedCustomer
+): customer is Stripe.Customer => {
+  if (typeof customer !== 'string') {
+    if (customer.deleted) {
+      return false;
+    } else {
+      return true;
+    }
+  } else {
+    return false;
+  }
+};
+
+const parseDryRun = (dryRun: boolean | string) => {
+  return `${dryRun}`.toLowerCase() !== 'false';
+};
+
+const parseDateForFirestore = (date: number) => Math.floor(date / 1000);
+
+const parseStartDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-23T00:00:00')
+  );
+
+const parseEndDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-31T23:59:59')
+  );
+
+const parseSubscriptionStatus = (
+  status: string
+): Stripe.Subscription['status'] | undefined => {
+  if (status) {
+    if (!isOfStatus(status)) {
+      throw new Error(
+        `Please provide a valid status. Invalid status: ${status}`
+      );
+    } else {
+      return status as Stripe.Subscription['status'];
+    }
+  } else {
+    return undefined;
+  }
+};
```

Comment:
Pretty sure we don't have https://eslint.org/docs/latest/rules/no-else-return enabled, but we might be able to reduce number of code branches by removing unneeded `else`s.

```js
const parseSubscriptionStatus = (
  status: string
): Stripe.Subscription['status'] | undefined => {
  if (status) {
    if (!isOfStatus(status)) {
      throw new Error(
        `Please provide a valid status. Invalid status: ${status}`
      );
    }
    return status as Stripe.Subscription['status'];
  }
  return undefined;
};
```